### DISCUSSION TITLE
Is it possible to customize the generated class names?

### DISCUSSION THREAD

**Discussion by jahirfiquitiva** [ben-rogerson/twin.macro#572] [General]
I've noticed generated class names have the component name as suffix (example: `css-abc123-StyledLink`) ... 

is it possible to setup a custom name for generated classes? maybe via a function that receives the component name as parameter and allows transforming it into the desired name?

If not, is it possible to include such functionality? maybe something like

```js
const cssPrefix = 'brand-name';
componentClassNameTransformer = (componentName, preGeneratedClassName) => {
    // componentName = StyledLink
    // preGeneratedClassName = css-abc123-StyledLink
    const [,uniqueClassName] = preGeneratedClassName.split('-')
    return `${cssPrefix}-${uniqueClassName}-${componentName.toLowerCase()}`;
}
```

then, the generated html would have class name: `brand-name-abc123-styledlink`

**Comment by ben-rogerson** [comment]
Twin doesn't control this part of the process, that's controlled by your css-in-js library.
If you're using styled-components you can set the [namespace property](https://styled-components.com/docs/tooling#namespace) and emotion has a [similar option](https://github.com/emotion-js/emotion/issues/2242).
